packages/packages/documents/src/store/sync.ts

```diff
@@ -0,0 +1,139 @@
+import { Document, Slice } from '../types';
+import { dispatch } from '@elementor/store';
+import { normalizeV1Document, getDocumentsManager } from './utils';
+import { commandEndEvent, CommandEvent, commandStartEvent, listenTo, v1ReadyEvent } from '@elementor/v1-adapters';
+
+export function syncStore( slice: Slice ) {
+	syncDocuments( slice );
+	syncCurrentDocument( slice );
+	syncOnDocumentSave( slice );
+	syncOnDocumentChange( slice );
+}
+
+function syncDocuments( slice: Slice ) {
+	const { setDocuments } = slice.actions;
+
+	listenTo(
+		v1ReadyEvent(),
+		() => {
+			const normalizedDocuments = Object
+				.entries( getDocumentsManager().documents )
+				.reduce( ( acc: Record<string, Document>, [ id, document ] ) => {
+					acc[ id ] = normalizeV1Document( document );
+
+					return acc;
+				}, {} );
+
+			dispatch( setDocuments( normalizedDocuments ) );
+		}
+	);
+}
+
+function syncCurrentDocument( slice: Slice ) {
+	const { setCurrentDocumentId, addDocument } = slice.actions;
+
+	listenTo(
+		[
+			v1ReadyEvent(),
+			commandEndEvent( 'editor/documents/open' ),
+		],
+		() => {
+			const documentsManager = getDocumentsManager();
+
+			dispatch( setCurrentDocumentId( documentsManager.getCurrentId() ) );
+			dispatch( addDocument( normalizeV1Document( documentsManager.getCurrent() ) ) );
+		}
+	);
+}
+
+function syncOnDocumentSave( slice: Slice ) {
+	const { updateDocument } = slice.actions;
+
+	const setIsSaving = ( isSaving: boolean ) => {
+		const id = getDocumentsManager().getCurrentId();
+
+		dispatch( updateDocument( {
+			id,
+			isSaving,
+		} ) );
```

Comment:
move `isSaving` to top level, and use standard actions.

Reply:
No